taking-turns.js

```diff
@@ -0,0 +1,209 @@
+      console.log(`I don't understand what you're telling me!`);
+      highOrLow = await ask(`Is it higher (H), or lower (L)? `);
+    }
+    
+    // update currLow or currHigh based on highOrLow
+    if (highOrLow.toUpperCase() == "H") {
+      if (currGuess == currHigh) {
+        console.log(`It can't be higher than ${currGuess}, earlier you said it was lower than ${currHigh+1}!`);
+        guessCount--;   // this adjusts for bad response
+      } else {
+        currLow = currGuess + 1; // since it is higher, add 1
+      }
+    // else highOrLow == "L"
+    } else {
+      if (currGuess == currLow) {
+        console.log(`It can't be lower than ${currGuess} if you said it was higher than ${currLow-1}`);
+        guessCount--;   // this adjusts for bad user response
+      } else {
+        currHigh = currGuess - 1; // since lower, subtract 1
+      }
+    }
+    
+    // recalculate guess, increment guessCount
+    // currGuess = randomInteger(currLow, currHigh);
+    currGuess = findMid(currLow, currHigh);
+
+    guessCount++;
+    
+    // ask player about currGuess
+    yesOrNo = await validateInput(await ask(`Is it... ${currGuess}? (y or n) `), currGuess);
+    
+  }
+
+  console.log(`Your number was ${currGuess}!`);
+  console.log(`I guessed it in ${guessCount} times!`);
+    
+  // exit the game
+  //process.exit();
+}
+
+
+// function containing the game logic for the user guessing
+async function userGuesses() {
+  // variables & constants
+  let MAX = process.argv[2] || 100;    // get max number from argv
+  let MIN = 0;
+  let currHigh = MAX;
+  let currLow = MIN;
+  let userGuess;
+  let numGuesses = 0;
+  let compNum;    // holds the computer's number
+  let found = false;
+
+  // log intro message
+  console.log(`I will pick a number between ${MIN} and ${MAX}.`);
+  console.log(`Try to guess my number in as little tries as possible!`);
+
+  // get random number for computer
+  compNum = randomInteger(MIN, MAX);
+
+  // main while loop
+  // while number is not found, user makes a guess
+  while (!found) {
+
+    // prompt user to guess
+    // parseInt from user input to get a number instead of a string
+    userGuess = parseInt(await ask(`Make a guess! `));
+    numGuesses++;   // increment number of guesses
+
+    // make sure user guesses a number
+    while (typeof(userGuess) !== 'number') {
+      console.log(`Enter a number please!`);
+      userGuess = parseInt(await ask(`Make a guess! `));
+    }
+
+    // make sure user guesses within the range
+    while (userGuess > currHigh || userGuess < currLow) {
+      if (userGuess > currHigh) console.log(`Hey, I already told you that my number is lower than ${currHigh + 1}!`);
+      if (userGuess < currLow) console.log(`Hey, I already told you my number is higher than ${currLow - 1}`);
+      userGuess = parseInt(await ask(`Make a guess! `));
+    }
+
+    // if user guesses number correctly
+    if (userGuess == compNum) {
```

Comment:
consider using === instead of == as triple equals requires a strict match whereas double equals only checks that they are equivalent